quasiquotes3/shared/src/test/scala/scala/meta/tests/SuccessSuite.scala

```diff
@@ -0,0 +1,2955 @@
+package scala.meta.tests
+package quasiquotes
+
+import munit._
+import scala.meta._
+import scala.meta.dialects.Scala211
+import scala.meta.quasiquotes._
+import munit.Clue.generate
+import scala.meta.trees._
+import meta.prettyprinters.XtensionStructure
+
+import compat.Platform.EOL
+
+import meta.prettyprinters.XtensionSyntax
+
+class Success extends TreeSuiteBase {
+  test("rank-0 liftables") {
+    assertTree(q"foo[${42}]")(Term.ApplyType(Term.Name("foo"), List(Lit.Int(42))))
+    assertTree(q"${42}")(Lit.Int(42))
+  }
+
+  test("rank-1 liftables") {
+    implicit def custom: Lift[List[Int], Term.ArgClause] =
+      Lift(lst => Term.ArgClause(lst.map(x => q"$x".asInstanceOf[Term])))
+    assertTree(q"foo(..${List(1, 2, 3)})")(
+      Term.Apply(Term.Name("foo"), List(Lit.Int(1), Lit.Int(2), Lit.Int(3)))
+    )
+  }
+
+  test("construction ascriptions") {
+    val xs = List(q"x", q"y").asInstanceOf[List[Term]]
+    assertEquals(q"foo(..${xs})".syntax, "foo(x, y)")
+    val xss = List(List(q"x", q"y")).asInstanceOf[List[List[Term]]]
+    assertEquals(q"foo(...${xss})".syntax, "foo(x, y)")
+    val rhs = q"x".asInstanceOf[Term]
+    assertEquals(q"var foo = ${rhs}".syntax, "var foo = x")
+  }
+
+  // test("deconstruction ascriptions") {
+  //   val q"foo(..${xs: List[Term]})" = q"foo(x, y)"
```

Comment:
Ach, I see where we have the type ascriptions, I wasn't even sure it was possible. Would be good to support but if it's too hard we can leave it out.

Reply:
without a new SIP it is impossible unfortunately

Reply:
It's fine to drop that functionality for Scala 2 then